scalafmt-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -1,25 +1,194 @@
+      case Configured.NotOk(e) => throw new MessageOnlyException(e.msg)
+    }).getOrElse(ScalafmtConfig.default))
+  private val sbtConfig = scalaConfig.map(_.forSbt)
+
+  private type Input = String
+  private type Output = String
+
+  private def withFormattedSources[T](
+      sources: Seq[File],
+      config: ScalafmtConfig
+  )(
+      onError: (File, Throwable) => T,
+      onFormat: (File, Input, Output) => T
+  ): Seq[Option[T]] = {
+    sources.map(
+      file => {
+        val input = IO.read(file)
+        val output = Scalafmt.format(input, config)
+
+        output match {
+          case Formatted.Failure(e) =>
+            if (config.runner.fatalWarnings) {
+              throw e
+            } else if (config.runner.ignoreWarnings) {
+              // do nothing
+              None
+            } else {
+              Some(onError(file, e))
+            }
+          case Formatted.Success(code) =>
+            Some(onFormat(file, input, code))
+        }
+      }
+    )
+  }
+
+  private def formatSources(
+      sources: Seq[File],
+      config: ScalafmtConfig,
+      log: Logger
+  ): Unit = {
+    withFormattedSources(sources, config)(
+      (file, e) => log.error(s"Error in ${file.toString}: $e"),
+      (file, input, output) => {
+        if (input != output) {
+          log.info(s"Successfully formatted ${file.toString}.")
+          IO.write(file, output)
+        }
+      }
+    )
+    PlatformTokenizerCache.megaCache.clear()
+  }
+
+  private def checkSources(
+      sources: Seq[File],
+      config: ScalafmtConfig,
+      log: Logger
+  ): Boolean = {
+    val res = withFormattedSources(sources, config)(
+      (file, e) => {
+        log.error(s"Error in ${file.toString}: $e")
+        false
+      },
+      (file, input, output) => {
+        val diff = input != output
+        if (diff) {
+          throw new MessageOnlyException(
+            s"${file.toString} isn't formatted properly!")
+        }
+        diff
       }
+    ).flatten.forall(x => x)
+    PlatformTokenizerCache.megaCache.clear()
+    res
   }
-  override def globalSettings: Seq[Def.Setting[_]] =
-    Seq(
-      commands += autoImport.scalafmt
-    ) ++
-      addCommandAlias("scalafmtTest", "scalafmt --test") ++
-      addCommandAlias("scalafmtDiffTest", "scalafmt --diff --test") ++
-      addCommandAlias("scalafmtDiff", "scalafmt --diff")
 
+  private lazy val sbtSources = thisProject.map(
+    proj => {
+      val rootSbt =
+        BuildPaths.configurationSources(proj.base).filterNot(_.isHidden)
+      val projectSbt =
+        (BuildPaths.projectStandard(proj.base) * GlobFilter("*.sbt")).get
+          .filterNot(_.isHidden)
+      rootSbt ++ projectSbt
+    }
+  )
+  private lazy val projectSources = thisProject.map(proj =>
+    (BuildPaths.projectStandard(proj.base) * GlobFilter("*.scala")).get)
+
+  lazy val scalafmtConfigSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmt := formatSources(
+      (unmanagedSources in scalafmt).value,
```

Comment:
This includes non-Scala source like *.java files. Running on the akka build results in a lot of parse errors like this

```scala
[error] Error in /Users/ollie/dev/akka/akka-testkit/src/test/java/akka/testkit/AkkaJUnitActorSystemResource.java: <input>:48: error: expected class or object definition
[error] public class AkkaJUnitActorSystemResource extends ExternalResource {
[error] ^
[error] Error in /Users/ollie/dev/akka/akka-testkit/src/test/java/akka/testkit/TestActorRefJavaCompile.java: <input>:10: error: expected class or object definition
[error] public class TestActorRefJavaCompile {
[error] ^
```

Reply:
We should add some *.java files to the scripted tests